ecdsa.go

```diff
@@ -300,3 +300,44 @@ func (c *Context) GenerateECDSAKeyPairWithAttributes(public, private AttributeSe
 func (signer *pkcs11PrivateKeyECDSA) Sign(rand io.Reader, digest []byte, opts crypto.SignerOpts) ([]byte, error) {
 	return signer.context.dsaGeneric(signer.handle, pkcs11.CKM_ECDSA, digest)
 }
+
+// Perform CKM_ECDH1_DERIVE function with the given public key
```

Comment:
Can you add some extra commentary here to help the user understand this function does not provide support for ANSI X9.63 ECDH derivation.

Reply:
i apologize - this is getting to a level of cryptography I'm not sure I can explain correctly. Maybe you can help me explain this nuance as I'm not sure I understand how it differs?

Reply:
not sure if it helps, but I used this logic as my guide:
 https://github.com/OpenSC/OpenSC/blob/master/src/tools/pkcs11-tool.c#L3693
and

Reply:
If you look at https://www.cryptsoft.com/pkcs11doc/STANDARD/pkcs-11v2-20.pdf section 12.3.8 the specification allows for 2 different KDF-defining parameters; `CKD_NULL` as used here and `CKD_ SHA1_KDF` which can be used to derive keys inline with the ANSI X9.63 spec. The comment simply needs to reflect that this function makes no provision to support ANSI X9.63 key derivation

Reply:
thanks for the help! i've force pushed a new version with an updated comment